File: programs/3-semantics+codegen/valid/8-12-forstmts.go
//~infinite
//~0
//~1
//~2
//~0
//~1
//~2
//~0
//~0
//~1
//~1
//~2
//~2
//~0
//~0
//~0
//~Success

package main

func main() {
	for {
		println("infinite")
		break
	}

	var x int  = 0
	for x < 3 {
		println(x)
		x++
	}

	for i :=0; i < 3; i++ {
		println(i)
	}

	for a,b := 0, 0; a < 3; b,a = a,b {
		println(a)
		a++
	}

	for a := 0; a < 3; a++ {
		var a int
		println(a)
	}

	for i := 0; i < 3; i++ {
		continue
		println(i)
	}
	println("Success")
}

File: grading/3-semantics+codegen/valid/8-12-forstmts.py
# import all necessary Python libraries
from __future__ import print_function

def main():
    while True:
        print(str("infinite").lower() if type("infinite") is bool else "infinite",  sep=' ', end='\n')
        break

    _GOLITE__x = 0
    while (_GOLITE__x < 3):
        print(str(_GOLITE__x).lower() if type(_GOLITE__x) is bool else _GOLITE__x,  sep=' ', end='\n')
        _GOLITE__x+= 1

    _GOLITE__i = 0
    while (_GOLITE__i < 3):
        print(str(_GOLITE__i).lower() if type(_GOLITE__i) is bool else _GOLITE__i,  sep=' ', end='\n')
        _GOLITE__i+= 1
    _GOLITE__a, _GOLITE__b = 0, 0
    while (_GOLITE__a < 3):
        print(str(_GOLITE__a).lower() if type(_GOLITE__a) is bool else _GOLITE__a,  sep=' ', end='\n')
        _GOLITE__a+= 1
        _GOLITE__b, _GOLITE__a = _GOLITE__a, _GOLITE__b
    _GOLITE__a = 0
    while (_GOLITE__a < 3):
        _GOLITE__a = 0
        print(str(_GOLITE__a).lower() if type(_GOLITE__a) is bool else _GOLITE__a,  sep=' ', end='\n')
        _GOLITE__a+= 1
    _GOLITE__i = 0
    while (_GOLITE__i < 3):
        _GOLITE__i+= 1
        continue
        print(str(_GOLITE__i).lower() if type(_GOLITE__i) is bool else _GOLITE__i,  sep=' ', end='\n')
    print(str("Success").lower() if type("Success") is bool else "Success",  sep=' ', end='\n')

if __name__ == "__main__":
    main()